util/econftool.c

```diff
@@ -0,0 +1,557 @@
+            fprintf(stdout, "Do you really want to overwrite it?\nYes [y], no [n]\n");
+            scanf("%2s", input);
+        } while (strcmp(input, "y") != 0 && strcmp(input, "n") != 0);
+
+        if (strcmp(input, "y") == 0) {
+            if ((econf_error = econf_writeFile(key_file_edit, path, filename_suffix))) {
+                fprintf(stderr, "%s\n", econf_errString(econf_error));
+                ret = -1;
+                goto cleanup;
+            }
+        }
+    } else {
+        if ((econf_error = econf_writeFile(key_file_edit, path, filename_suffix))) {
+            fprintf(stderr, "%s\n", econf_errString(econf_error));
+            econf_free(key_file_edit);
+            ret = -1;
+            goto cleanup;
+        }
+    }
+
+    /* cleanup */
+    cleanup:
+    econf_free(key_file_edit);
+    econf_free(groupsEdit);
+    econf_free(groups);
+    return ret;
+
+}
+
+/**
+ * @biref Revert all changes to the vendor version. If the user is root,
+ *        delete all files in /etc for config file including snippet directory.
+ *        If the user is not root delete the config file in xdg_config_dir.
+ */
+static int econf_revert(bool is_root)
+{
+    char input[3] = "";
+    int status = 0;
+
+    if (!is_root) {
+        snprintf(path_filename, sizeof(path_filename), "%s/%s", xdg_config_dir, filename_suffix);
+    } else {
+        snprintf(path_filename, sizeof(path_filename), "/etc/%s", filename_suffix);
+    }
+
+    if (access(path_filename, F_OK) == -1 && (access(path, F_OK) == -1 || !is_root)) {
+        fprintf(stderr, "Could not find %s or a snippet directory\n", path_filename);
+        return -1;
+    }
+
+    /* Delete config file */
+    if (access(path_filename, F_OK) == 0) {
+        /* let the user verify that the file should really be deleted */
+        while (strcmp(input, "y") != 0 && strcmp(input, "n") != 0
+                && !non_interactive) {
+            fprintf(stdout, "Delete file %s?\nYes [y], no [n]\n", path_filename);
+            scanf("%2s", input);
+        }
+
+        if (strcmp(input, "y") == 0 || non_interactive) {
+            status = remove(path_filename);
+            if (status)
+                perror("remove() failed");
+            else
+                fprintf(stdout, "File %s deleted!\n", path_filename);
+        }
+    }
+
+    if (!is_root)
+        return 0;
+
+    /* Reset input */
+    strcpy(input, "");
+
+    /* Delete snippet directory */
+    if (access(path, F_OK) == 0) {
+        /* let the user verify that the directory should really be deleted */
+        while (strcmp(input, "y") != 0 && strcmp(input, "n") != 0 && !non_interactive) {
+            fprintf(stdout, "Delete directory %s?\nYes [y], no [n]\n", path);
+            scanf("%2s", input);
+        }
+
+        if (strcmp(input, "y") == 0 || non_interactive) {
+            status = nftw(path, nftw_remove, getdtablesize() - 10, FTW_DEPTH);
```

Comment:
Don't follow symlinks:

```suggestion
            status = nftw(path, nftw_remove, getdtablesize() - 10, FTW_DEPTH | FTW_PHYS);
```